### PR TITLE
chore(homepage): optimize generation and add test

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**

⚒️ Fixes  # 

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [x] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: library/common/templates/lib/ingress/integrations/_homepage.tpl [hunk 1]
```diff
@@ -9,13 +9,14 @@
...
-    {{- $svcData := (include "tc.v1.common.lib.ingress.targetSelector" (dict "rootCtx" $rootCtx "objectData" $objectData) | fromYaml) -}}
 
     {{- $name := $homepage.name | default ($rootCtx.Chart.Name | camelcase) -}}
     {{- $desc := $homepage.description | default $rootCtx.Chart.Description -}}
     {{- $icon := $homepage.icon | default $rootCtx.Chart.Icon -}}
-    {{- $type := ( $homepage.widget.type | default $rootCtx.Chart.Name ) | lower -}}
-    {{- $type = regexReplaceAll "\\W+" $type "" -}}
+    {{- $defaultType := $rootCtx.Chart.Name -}}
+    {{/* Remove any non-characters from the default type */}}
+    {{- $defaultType = regexReplaceAll "\\W+" $type "" -}}
```
Comment (stavros-k) [inline library/common/templates/lib/ingress/integrations/_homepage.tpl:18]: Only modifies the default one, so user's define are untouched

File: library/common/templates/lib/ingress/_targetSelector.tpl [hunk 2]
```diff
@@ -79,7 +79,12 @@
     {{- $expandedSvcName = printf "%s-%s" $expandedSvcName $svcName -}}
   {{- end -}}
 
-  {{- $selectedService = (dict "name" $expandedSvcName "port" (tpl ($portData.port | toString) $rootCtx)) -}}
+  {{- $protocol := default "http" -}}
+  {{- if eq $portData.protocol "https" -}}
+    {{- $protocol = "https" -}}
+  {{- end -}}
+
+  {{- $selectedService = (dict "name" $expandedSvcName "port" (tpl ($portData.port | toString) $rootCtx) "protocol" $protocol) -}}
```
Comment (stavros-k) [inline library/common/templates/lib/ingress/_targetSelector.tpl:87]: Also handles protocol of the port

File: library/common/templates/class/_ingress.tpl [hunk 3]
```diff
@@ -12,6 +12,7 @@ objectData: The object data to be used to render the Ingress.
   {{- $objectData := .objectData -}}
 
   {{- $svcData := (include "tc.v1.common.lib.ingress.targetSelector" (dict "rootCtx" $rootCtx "objectData" $objectData) | fromYaml) -}}
+  {{- $_ := set $objectData "selectedService" $svcData -}}
```
Comment (stavros-k) [inline library/common/templates/class/_ingress.tpl:15]: Its a slow operation, lets "cache" it and reuse the data in the integration.

Comment (stavros-k) [reply library/common/templates/class/_ingress.tpl:15]: Tho we might want to take into account the overrideService?

Comment (Ornias1993) [reply library/common/templates/class/_ingress.tpl:15]: we can, but I think that gets very messy, but we can.

Comment (stavros-k) [reply library/common/templates/class/_ingress.tpl:15]: Oh wait we cant, override is per host and per path.
I don't want to start picking random ones.
If a user starts overriding stuff, the user can also override the url too

File: library/common/templates/lib/ingress/integrations/_homepage.tpl [hunk 4]
```diff
@@ -29,11 +30,12 @@
...
-      {{- $svc := $svcData.name -}}
-      {{- $port := $svcData.port -}}
-      {{- $ns := printf "%s" (include "tc.v1.common.lib.metadata.namespace" (dict "rootCtx" $rootCtx "objectData" $objectData "caller" "Ingress")) -}}
+      {{- $svc := $objectData.selectedService.name -}}
+      {{- $port := $objectData.selectedService.port -}}
+      {{- $prot := $objectData.selectedService.protocol -}}
+      {{- $ns := include "tc.v1.common.lib.metadata.namespace" (dict "rootCtx" $rootCtx "objectData" $objectData "caller" "Ingress") -}}
 
-      {{- $url = printf "http://%s.$ns.svc:%s" $svc $ns $port -}}
+      {{- $url = printf "%s://%s.%s.svc:%s" $prot $ns $svc $port -}}
```
Comment (stavros-k) [inline library/common/templates/lib/ingress/integrations/_homepage.tpl:38]:
```suggestion
      {{- $url = printf "%s://%s.%s.svc:%s" $prot $svc $ns $port -}}
```

**Pull request merged**